sdk/core/azure-core/inc/azure/core/response.hpp

```diff
@@ -95,4 +95,36 @@ namespace Azure { namespace Core {
       return std::move(this->m_rawResponse);
     }
   };
+
+  /**
+   * @brief Wraps raw HTTP response into a response of a void type.
+   */
+  template <> class Response<void> {
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    /**
+     * @brief Initialize a #Response<void> with a raw response.
+     *
+     * @param rawResponse Raw HTTP response.
+     */
+    explicit Response(std::unique_ptr<Http::RawResponse>&& rawResponse)
+        : m_rawResponse(std::move(rawResponse))
+    {
+    }
+
+    /**
+     * @brief Get raw HTTP response.
+     */
+    // Do not give up raw response ownership.
+    Http::RawResponse& GetRawResponse() { return *this->m_rawResponse; }
+
+    /**
+     * @brief Get a smart pointer rvalue reference to the value of a specific type.
```

Comment:
I am not sure how user-friendly or relevant terms like `rvalue reference` are to the end user, even in the context of C++.

Would something like this be better?
```suggestion
     * @brief Get a unique pointer to the raw response and take ownership.
```

Reply:
I just realized this is similar to the existing docs. We can address this separately and fix them all.

It was added in https://github.com/Azure/azure-sdk-for-cpp/pull/611 so the wording isn't set in stone.

Reply:
I copied the comments from the template one. https://github.com/Azure/azure-sdk-for-cpp/blob/5dce1796b351ca0f0e6e4156354c2081ddd3316a/sdk/core/azure-core/inc/azure/core/response.hpp#L86

I think it will be good to be consistent.


As for your suggestion, I think it's worth pointing out that this function returns an rvalue reference, because we got another function `GetRawResponse` which returns an lvalue reference (which by the way I think should return a const lvalue reference, but that's not related to this PR).